### PR TITLE
Enable CLI configuration of Version and Commandline switches via plugin hooks

### PR DESCRIPTION
**Pull request by SidneyS**
Hi,

I came up with a practical - as in, plugin-agnostic and hopefully timesaving -  polyfill for an upcoming feaure in cordova-lib, which lets us specify plugin preference defaults and enables supplying CLI variables with hard-requiring them (see https://issues.apache.org/jira/browse/CB-9162).

Motivation? The programmatic setup of Crosswalk plugin is often required by 3rd party CLIs or other tools, which defer to or embed cordova. (e.g., Ionic via "ionic browser add").

Need XWalk 15 for a new project and want to test experimental Chromium features? Try:
```bash
cordova plugin add cordova-plugin-crosswalk-webview --variable CROSSWALK_ANDROID_VERSION="15" --variable CROSSWALK_ANDROID_COMMANDLINE="--enable-usermedia-screen-capturing"
```
Of course, all present configuration vectors remain active. The hook simply gets its defaults by parsing plugin.xml, and creates configuration elements in config.xml (which are in turn parsed by platforms). Full JSDoc documentation inside.

Also did some very minor cleanup. Looking forward to your input.

Sincerely,
S

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -12,22 +12,35 @@
...
+        <source-file src="platforms/android/src/org/crosswalk/engine/XWalkExposedJsApi.java" target-dir="src/org/crosswalk/engine"/>
+        <source-file src="platforms/android/src/org/crosswalk/engine/XWalkCordovaResourceClient.java" target-dir="src/org/crosswalk/engine"/>
+        <source-file src="platforms/android/src/org/crosswalk/engine/XWalkCordovaUiClient.java" target-dir="src/org/crosswalk/engine"/>
+        <source-file src="platforms/android/src/org/crosswalk/engine/XWalkCordovaView.java" target-dir="src/org/crosswalk/engine"/>
+        <source-file src="platforms/android/src/org/crosswalk/engine/XWalkCordovaCookieManager.java" target-dir="src/org/crosswalk/engine"/>
+
+        <framework src="platforms/android/xwalk.gradle" custom="true" type="gradleReference"/>
 
-        <framework src="src/android/xwalk.gradle" custom="true" type="gradleReference" />
+        <hook type="after_plugin_install" src="hooks/after_plugin_install/000-apply_cli_options.js"/>
```
Comment (fujunwei) [inline plugin.xml]: Cordova-lib/cli 5.2.0 will release recently. is the plugin still need the hook?

File: platforms/android/xwalk.gradle [hunk 2]
```diff
@@ -51,10 +51,10 @@ def getConfigPreference(name, defaultValue) {
 
 // Set defaults before project's build-extras.gradle
 if (!project.hasProperty('xwalkVersion')) {
-    ext.xwalkVersion = getConfigPreference("xwalkversion", DEFAULT_VERSION)
+    ext.xwalkVersion = getConfigPreference("CROSSWALK_ANDROID_VERSION", DEFAULT_VERSION)
```
Comment (fujunwei) [inline platforms/android/xwalk.gradle]: The README also need align with it.
xwalkVersion --------->  CROSSWALK_ANDROID_VERSION
xwalkCommandLine ---------> CROSSWALK_ANDROID_COMMANDLINE

Comment (SidneyS) [inline platforms/android/xwalk.gradle]: Hi @fujunwei, i just added the readme fixes and updated the PR.

Thanks!

File: plugin.xml (hunk 1 above)
Comment (fujunwei) [inline plugin.xml]: Make sense. it will be removed when the feature is available in cordova-lib stable.

**Comment by fujunwei** [comment]
LGTM. ping @lincsoon @huningxin

**Comment by huningxin** [comment]
LGTM. Thanks!

**Pull request merged**